pkg/auth/handler/change_password_test.go

```diff
@@ -149,13 +149,14 @@ func TestChangePasswordHandler(t *testing.T) {
 			So(resp.Body.Bytes(), ShouldEqualJSON, `
 				{
 					"error": {
-						"code": 111,
-						"name": "PasswordPolicyViolated",
-						"message": "password too short",
+						"name": "Invalid",
+						"reason": "PasswordPolicyViolated",
+						"message": "password policy violated",
+						"code": 400,
 						"info": {
-							"reason": "PasswordTooShort",
-							"min_length": 6,
-							"pw_length": 4
+							"causes": [
+								{ "kind": "PasswordTooShort", "min_length": 6, "pw_length": 4 }
+							]
```

Comment:
Currently module defines its own shape of info, I think it is good that we have more flexible for future features. But seems now we already have some conventions?

For errors that may has multiple causes:
`{ "causes": [{"kind": "PasswordTooShort"}] }`
For error that only has single cause: 
`{ "cause": "ExpiredCode" }`

When new developers develop gears, they may not know this convention when developing new feature. To avoid having inconsistent in future maybe we can 1. Have helper functions to create cause and causes info. or  2. We have documentation about the error conventions. I prefer 1, thought?

Also for the shape of single cause, would it be better to use object instead of string? So we can add more data about the error if needed?

e.g.
```
{ "cause": { "kind": "ExpiredCode" } }
```

Reply:
Yes, the `causes` & `cause` is explained in the issue's note. I intended to organize the notes into the repo doc after reviewing.

I don't think a helper function would be much useful, since not all error would use this 'cause' convention:
- Some errors are not precisely an error, such as authentication session. The token and step is put in info directly.
- Some error would only have single kind of cause, such as user disabled error. In this case, the extra info is put directly in info.

For single cause error, I would prefer to flatten it directly at the info (i.e. `cause` field is same as `kind`). For example:
```json
{ "cause": "ExpiredCode", "user_id": "..." }
```

Reply:
Just my gut feeling, `causes` and `cause` would make people think their difference is just cardinality. If the type of `cause` is `T`, then `causes` should be of type `T[]`. Currently it is not the case.

Reply:
That's a good point. I'll add helper functions `NewWithCause`/`NewWithCauses`, which helps to populate the info in proper format. However, I'll keep the `NewWithDetails` function for the non-conforming errors mentioned above.